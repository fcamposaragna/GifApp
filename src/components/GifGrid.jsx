import PropTypes from 'prop-types';

import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

    const { images, isLoading} = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading && <h2>Cargando...</h2> 
            }
            <div className="card-grid">
                {
                    images.map(({ id, url, title}) => 
                        (
                           <GifItem key={ id }
                           url={ url }
                           title={ title }
                           />
                        )
                    )
                }
            </div>
        </>
    )
}

GifGrid.proptypes = {
    category: PropTypes.string.isRequired
}