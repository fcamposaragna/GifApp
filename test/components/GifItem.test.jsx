import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => {

    const title = 'Ricky and Morty';
    const url = "http://localhost/www.url.com";

    test('Debe de hacer match con el snapshot', ()=>{
        const {container} = render(<GifItem 
            title= { title }
            url= { url }/>);

        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => { 

        render( <GifItem title={title} url={url} />);
        //screen.debug();
        //expect(screen.getByRole('img').src).toBe( url );
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe de mostrar el tÃ­tlo en el componente', () => {

        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

    });

})