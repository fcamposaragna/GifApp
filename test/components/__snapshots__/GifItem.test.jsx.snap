// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Ricky and Morty"
      src="http://localhost/www.url.com"
    />
    <p>
      Ricky and Morty
    </p>
  </div>
</div>
`;
